File: leetcode3/황은지/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let max=0;
    let maxProfit=0;
    for(let i=prices.length-1;i>=0;i--){
        max=Math.max(max,prices[i]);
        maxProfit=Math.max(maxProfit,max-prices[i]);
    }

    return maxProfit;
};

File: leetcode3/남효정/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        small = prices[0]
        
        for i in range(len(prices)):
            profit = prices[i] - small
            ans = max(ans, profit)
            small = min(small, prices[i])

        return ans

        